### PR TITLE
rocket reach action

### PR DESCRIPTION
**Pull request by Jordon-L**
The `props` are all labeled as optional because the endpoints allow for multiple different parameters. For lookup profile specifically, inputting only the name prop or only the current employer prop will result in an error, and Rocket Reach API will ask for more information. I have put in if statements to notify the user if one of these scenarios happens.

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by Jordon-L** [comment]
fixes #2461

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Review by vellames-turing — DISMISSED** [review]
Hey, thanks for your PR.
Could you please add the `package` files for this app?
Also, some props like `lindedinUrl` are being repeated, can you move it to the `app` and use propDefinitions to import it into the files?
It would be great to move the `axios` request to `app` as well. Please take a look at `figma` app and cations. You can follow that as a template. There you will find examples of all suggestions.
If you have any questions, please message me here. Me or another review will help you :)

**Comment by Jordon-L** [comment]
I don't have any `package` files. Would doing an `npm init` then following the format of the `package` files for `figma` be good?

**Comment by Jordon-L** [comment]
Do you need the `package-lock.json` too? My `npm` version is `8.10.0`. Is there going to be a compatibility issue?

**Comment by Jordon-L** [comment]
I have installed an older version of `npm`. my `package` files look the same as `figma`'s  `package` files now.

**Comment by Jordon-L** [comment]
Hi, I have added the suggestions. Let me know if I am missing anything or need to change anything.

File: components/rocketreach/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@pipedream/rocketreach",
+  "version": "0.0.1",
+  "description": "Pipedream RocketReach Components",
+  "main": "rocketreach.app.mjs",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (lcaresia) [inline components/rocketreach/package.json:8]:
```suggestion
```

File: components/rocketreach/rocketreach.app.mjs [hunk 2]
```diff
@@ -1,11 +1,51 @@
...
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getBaseUrl() {
+      return "https://api.rocketreach.co/v2/api";
+    },
+    _getHeaders() {
+      return {
+        "Content-Type": "application/json",
+        "Api-Key": `${this.$auth.api_key}`,
```
Comment (lcaresia) [inline components/rocketreach/rocketreach.app.mjs:25]: Create and call a getter here.

E.g.

```suggestion
        "Api-Key": `${this._apiKey()}`,
```

File: components/rocketreach/rocketreach.app.mjs [hunk 3]
```diff
@@ -1,11 +1,51 @@
...
+      const response = {
+        ...opts,
+        url: this._getBaseUrl() + opts.path,
+        headers: this._getHeaders(),
+      };
+      return response;
+    },
+    async lookupProfile(params, ctx = this) {
+      return axios(ctx, this._getAxiosParams({
+        method: "GET",
```
Comment (lcaresia) [inline components/rocketreach/rocketreach.app.mjs:38]: Method `GET` is default in `axios`, don't need put here.

**Review by lcaresia — DISMISSED** [review]
Thanks for you contribution.

I requested just some minor changes.

**Comment by Jordon-L** [comment]
Thank you for the feedback. I have applied all the suggestions.

**Comment by Jordon-L** [comment]
I am failing some pull request checks, Is there a problem with the code I submitted? This was not happening 2 weeks ago.

**Comment by dylburger** [comment]
@Jordon-L you'll want to update your fork with the latest `master` branch of the `pipedream` repo, then run:

```bash
git fetch origin
git merge origin/master

# from repo root
npm ci
npx pnpm install -r
git add pnpm-lock.yaml
git commit
git push
```

The error just means that the lock file for `pnpm` is out-of-date, and you need to pull in the latest changes.

**Comment by Jordon-L** [comment]
When I type in `npx pnpm install -r`, I get an error ` ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies`. Should I be concerned?

**Comment by dylburger** [comment]
> When I run `npx pnpm install -r`, I get an error ` ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies`. Should I be concerned?

No, that's fine. In the main repo we have a new `.npmrc` file that skips that check:

```bash
❯ cat .npmrc
auto-install-peers=true
```

**Review by dylburger — APPROVED** [review]

**Comment by dylburger** [comment]
@Jordon-L I reviewed all of your most-recent code and approved the changes. Merging them now.

**Comment by dylburger** [comment]
Looks like they published successfully!

**Comment by Jordon-L** [comment]
@dylburger Thank you for the feedback and help regarding the pull request errors.

**Pull request merged**